DocumentDataAPI/DocumentDataAPI/Controllers/SimilarDocumentController.cs

```diff
@@ -0,0 +1,127 @@
+    /// </summary>
+    /// <response code="200">Success: ID of the MainDocument.</response>
+    /// <response code="500">Internal Server Error: a <see cref="ProblemDetails"/> describing the error.</response>
+    [HttpPost]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<List<long>>> SetSimilarDocument([FromBody] List<SimilarDocumentModel> mainDocument)
+    {
+        try
+        {
+            IEnumerable<long> insertedIds = await _repository.AddBatch(mainDocument);
+            return Ok(insertedIds);
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Unable to add mainDocument.");
+            return Problem(e.Message);
+        }
+    }
+
+    /// <summary>
+    /// Retrieves a list of all similar documents from the database.
+    /// </summary>
+    /// <param name="limit">The maximum number of rows to get.</param>
+    /// <param name="offset">The number of rows to skip (previous offset + previous limit).</param>
+    /// <response code="200">Success: A list of all similar documents</response>
+    /// <response code="204">No Content: Nothing is returned.</response>
+    /// <response code="500">Internal Server Error: a <see cref="ProblemDetails"/> describing the error.</response>
+    [HttpGet]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<IEnumerable<SimilarDocumentModel>>> GetAll(int? limit = 100, int? offset = null)
+    {
+        try
+        {
+            IEnumerable<SimilarDocumentModel> result = await _repository.GetAll(limit, offset);
+            return result.Any()
+                ? Ok(result)
+                : NoContent();
+        }
+        catch (Exception e)
+        {
+            _logger.LogError(e, "Unable to get sources");
+            return Problem(e.Message);
+        }
+    }
+
+    /// <summary>
+    /// Retrieves a list of similar document based on the mainDocument id.
+    /// </summary>
+    /// <response code="200">Success: A list of similar document for the given mainDocument id.</response>
+    /// <response code="204">No Content: Nothing is returned.</response>
+    /// <response code="500">Internal Server Error: a <see cref="ProblemDetails"/> describing the error.</response>
+    [HttpGet]
+    [Route("/{mainDocumentId:long}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public async Task<ActionResult<IEnumerable<SimilarDocumentModel>>?> GetById(long mainDocumentId)
+    {
+        try
+        {
+            IEnumerable<SimilarDocumentModel>? result = await _repository.Get(mainDocumentId);
```

Comment:
```suggestion
            IEnumerable<SimilarDocumentModel> result = await _repository.Get(mainDocumentId);
```
Hvis Get returnere en tom liste, så skal der ikke være en nullable type her, da i bruger .any() herefter, hvori der er en mulig null reference.